tests/cnf/core/network/metallb/tests/bgp-remote-as-dynamic.go

```diff
@@ -119,6 +121,74 @@ var _ = Describe("BGP remote-dynamicAS", Ordered, Label(tsparams.LabelDynamicRem
+				By("Removing static routes from the speakers")
+				frrk8sPods, err := pod.List(APIClient, NetConfig.MlbOperatorNamespace, metav1.ListOptions{
+					LabelSelector: tsparams.FRRK8sDefaultLabel,
+				})
+				Expect(err).ToNot(HaveOccurred(), "Failed to list pods")
+
+				speakerRoutesMap := buildRoutesMapWithSpecificRoutes(frrk8sPods, []string{ipv4metalLbIPList[0],
+					ipv4metalLbIPList[1], frrNodeSecIntIPv4Addresses[0], frrNodeSecIntIPv4Addresses[1]})
+
+				for _, frrk8sPod := range frrk8sPods {
+					out, err := frr.SetStaticRoute(frrk8sPod, "del", frrExternalMasterIPAddress, speakerRoutesMap)
+					Expect(err).ToNot(HaveOccurred(), out)
+				}
+
+				By("Clean metallb operator and test namespaces")
+				resetOperatorAndTestNS()
+			})
+
+			It("Verify the establishment of a multi-hop iBGP adjacency using neighbor peer remote-as external",
+				reportxml.ID("76823"), func() {
+					hubPodWorkerNames := []string{hubPodNameWorker0, hubPodNameWorker1}
+					frrPod, frrk8sPods := setupBGPRemoteASMultiHopTest(ipv4metalLbIPList, hubIPv4ExternalAddresses,
+						externalAdvertisedIPv4Routes, externalAdvertisedIPv6Routes, hubPodWorkerNames,
+						frrExternalMasterIPAddress, tsparams.LocalBGPASN, false)
+
+					By("Creating a BGP Peer with dynamicASN")
+					createBGPPeerWithDynamicASN(frrExternalMasterIPAddress, dynamicASiBGP, false)
+					By("Checking that BGP session is established and up")
+					verifyMetalLbBGPSessionsAreUPOnFrrPod(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
+
+					By("Validating external FRR AS number received on the FRR nodes")
+					Eventually(func() error {
+						return frr.ValidateBGPRemoteAS(frrk8sPods, frrExternalMasterIPAddress, tsparams.LocalBGPASN)
+					}, 60*time.Second, 5*time.Second).Should(Succeed(),
+						fmt.Sprintf("The remoteASN does not match the expected AS: %d", tsparams.LocalBGPASN))
+				})
+
+			It("Verify the establishment of a multi-hop eBGP adjacency using neighbor peer remote-as external",
+				reportxml.ID("76824"), func() {
+					hubPodWorkerName := []string{hubPodNameWorker0, hubPodNameWorker1}
+					frrPod, frrk8sPods := setupBGPRemoteASMultiHopTest(ipv4metalLbIPList, hubIPv4ExternalAddresses,
+						externalAdvertisedIPv4Routes, externalAdvertisedIPv6Routes, hubPodWorkerName,
+						frrExternalMasterIPAddress, tsparams.RemoteBGPASN, true)
+
+					By("Creating a BGP Peer with dynamicASN")
+					createBGPPeerWithDynamicASN(frrExternalMasterIPAddress, dynamicASeBGP, true)
+
+					By("Checking that BGP session is established and up")
+					verifyMetalLbBGPSessionsAreUPOnFrrPod(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
+
+					By("Validating external FRR AS number received on the FRR nodes")
+					Eventually(func() error {
+						return frr.ValidateBGPRemoteAS(frrk8sPods, frrExternalMasterIPAddress, tsparams.RemoteBGPASN)
+					}, 60*time.Second, 5*time.Second).Should(Succeed(),
+						fmt.Sprintf("The remoteASN does not match the expected AS: %d", tsparams.RemoteBGPASN))
+				})
```

Comment:
just advice. Both tests are pretty similar only diff is BGPASN. You can create the local func that running all the test manupaltion - for ex. multihopTest (bgpASN  string). And call only this func uder IT with differen BGPASN

Reply:
I already created a local func setupBGPRemoteASMultiHopTest with a variable asNumber. I did not include the creation of the bgppeer because that is the actual test. I think its enough. If you have a specific suggestion lets me know. Thanks!